src/check_presence.c

```diff
@@ -0,0 +1,533 @@
+       *
+       */
+
+      const char *host = key.mv_data;
+
+      struct timeval *tv_then = {0};
+
+      tv_then = data.mv_data;
+
+      if (tv_now.tv_sec - tv_then->tv_sec >= host_timeout + thread_interval) {
+
+        char message[NOTIF_MAX_MSG_LEN];
+
+        rc = mdb_cursor_del(cursor, 0);
+        if (rc) {
+          ERROR(PLUGIN_NAME " plugin: mdb_cursor_del returned: %s (%d)",
+                mdb_strerror(rc), rc);
+          mdb_cursor_close(cursor);
+          mdb_txn_abort(txn);
+          goto skip;
+        }
+
+        ssnprintf(
+            message, sizeof(message), "Host not seen for %d seconds: %.*s",
+            (int)(tv_now.tv_sec - tv_then->tv_sec), (int)sizeof(n.host), host);
+        notification_init(&n, NOTIF_FAILURE, message, host, PLUGIN_NAME, NULL,
+                          "host", "lost");
+        n.time = cdtime();
+
+        plugin_dispatch_notification(&n);
+      }
+    }
+    mdb_cursor_close(cursor);
+
+    rc = mdb_txn_commit(txn);
+    if (rc) {
+      ERROR(PLUGIN_NAME " plugin: mdb_txn_commit returned: %s (%d)",
+            mdb_strerror(rc), rc);
+      goto skip;
+    }
+
+  skip:
+    pthread_mutex_lock(&presence_lock);
+
+    if (presence_thread_loop <= 0)
+      break;
+
+    ts_wait.tv_sec = tv_now.tv_sec + thread_interval;
+    ts_wait.tv_nsec = 0;
+
+    pthread_cond_timedwait(&presence_cond, &presence_lock, &ts_wait);
+    if (presence_thread_loop <= 0)
+      break;
+  } /* while (presence_thread_loop > 0) */
+
+  pthread_mutex_unlock(&presence_lock);
+
+  return (void *)0;
+} /* }}} void *presence_thread */
+
+static int start_thread(void) /* {{{ */
+{
+
+  pthread_mutex_lock(&presence_lock);
+
+  if (presence_thread_loop != 0) {
+    pthread_mutex_unlock(&presence_lock);
+    return 0;
+  }
+
+  presence_thread_loop = 1;
+  int status = plugin_thread_create(&presence_thread_id, /* attr = */ NULL,
+                                    presence_thread,
+                                    /* arg = */ (void *)0, "host");
+  if (status != 0) {
+    presence_thread_loop = 0;
+    ERROR(PLUGIN_NAME " plugin: starting thread failed.");
+    pthread_mutex_unlock(&presence_lock);
+    return -1;
+  }
+
+  pthread_mutex_unlock(&presence_lock);
+  return 0;
+} /* }}} int start_thread */
+
+static int stop_thread(void) /* {{{ */
+{
+
+  pthread_mutex_lock(&presence_lock);
+
+  if (presence_thread_loop == 0) {
+    pthread_mutex_unlock(&presence_lock);
+    return -1;
+  }
+
+  presence_thread_loop = 0;
+  pthread_cond_broadcast(&presence_cond);
+  pthread_mutex_unlock(&presence_lock);
+
+  int status = pthread_join(presence_thread_id, /* return = */ NULL);
+  if (status != 0) {
+    ERROR(PLUGIN_NAME " plugin: stopping thread failed.");
+    status = -1;
+  }
+
+  pthread_mutex_lock(&presence_lock);
+  memset(&presence_thread_id, 0, sizeof(presence_thread_id));
+  pthread_mutex_unlock(&presence_lock);
+
+  return status;
+} /* }}} int stop_thread */
+
+static int presence_init(void) {
```

Comment:
There is no way to check if your plugin working or not (i.e is configured or not) w/o doing "active tests".
I think that would be good to notify user "plugin is not configured and will not do any checks /send any notifications" (please rephrase!!!) if `state_datastore == NULL`.

Reply:
Done.